Assets/MirageXR/ContentTypes/Audio/Editors/Mobile/AudioCaptionGenerator.cs

```diff
@@ -0,0 +1,121 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using TMPro;
+using IBM.Cloud.SDK;
+using IBM.Cloud.SDK.Authentication.Iam;
+using IBM.Cloud.SDK.Utilities;
+using IBM.Cloud.SDK.DataTypes;
+using IBM.Watson.SpeechToText.V1;
+using IBM.Watson.SpeechToText.V1.Model;
+using System.IO;
+using UnityEngine.UI;
+
+public class AudioCaptionGenerator : MonoBehaviour
+{ 
+    [SerializeField] private TMP_Text transcribedSpeech;
+    
+    [SerializeField] private GameObject _spinner;
+    [SerializeField] private AudioEditorView _audioEditView;
+
+    private SpeechToTextService speechToText;
+    private string soundPath;
+    private string text = string.Empty;
+    private string _serviceUrl;
+    private string _iamApikey;
+
+    private AudioSource audioSource;
+    byte[] audioBytes;
+
+    AudioClip speechClip; // The AudioClip of the speech
+
+    void Start()
+    {
+        _iamApikey = _audioEditView._iamApikey_();
+        _serviceUrl = _audioEditView._serviceUrl_();
+         
+
+    StartCoroutine(CreateService());
+
+        if(_audioEditView.SaveAndReturnAudioClipPath() != null)
+        {
+            string audioFileName = _audioEditView.SaveAndReturnAudioClipPath();
+            soundPath = Path.Combine(audioFileName);
+            StartCoroutine(ConvertAudioFileToText(soundPath)); 
+        }
+    }
+
+    
+
+    private IEnumerator CreateService()
+    {
+        var authenticator = new IamAuthenticator(apikey: _iamApikey);
+
+        while (!authenticator.CanAuthenticate())
+            yield return null;
+
+        speechToText = new SpeechToTextService(authenticator);
+        speechToText.SetServiceUrl(_serviceUrl);
+    }
+
+    private IEnumerator ConvertAudioFileToText(string audioFilePath)
```

Comment:
it would be good to split this method into two. one translates audio into text, the other inserts this text into a UI element